src/posix/platform/radio_spinel.hpp

```diff
@@ -399,6 +399,38 @@ class RadioSpinel
     void Update(struct timeval &aTimeout);
 #endif
 
+#if OPENTHREAD_ENABLE_DIAG
+    /**
+     * This method enables/disables the factory diagnostics mode.
+     *
+     * @param[in]  aMode  TRUE to enable diagnostics mode, FALSE otherwise.
+     *
+     */
+    void SetDiagMode(bool aMode) { mDiagMode = aMode; }
```

Comment:
nit: how about `SetDiagEnabled()` and `IsDiagEnabled()` to better indicate this is a boolean?